auto_gptq/nn_modules/fused_llama_attn.py

```diff
@@ -150,41 +177,63 @@ def inject_to_model(
             k_proj = m.k_proj
             v_proj = m.v_proj
 
-            qweights = torch.cat([q_proj.qweight, k_proj.qweight, v_proj.qweight], dim=1)
-            qzeros = torch.cat([q_proj.qzeros, k_proj.qzeros, v_proj.qzeros], dim=1)
-            scales = torch.cat([q_proj.scales, k_proj.scales, v_proj.scales], dim=1)
-
-            if QuantLinear.QUANT_TYPE == "exllama":
-                if desc_act:
-                    # TODO: support it. The issue lies maybe in the line:
-                    # int groups = qzeros.size(0);
-                    # in exllama_ext.cpp
-                    raise ValueError("Exllama kernel does not support query/key/value fusion with act-order. Please either use inject_fused_attention=False or disable_exllama=True.")
-                else:
+            if QuantLinear.QUANT_TYPE == "exllama" and desc_act:
+                qkv_layers = [q_proj,k_proj,v_proj]
```

Comment:
Is this really working? Did you verify correctness? I was thinking it is more of an issue in the CUDA kernel.